### ISSUE TITLE
mnt: Can't mount at ./etc/resolv.conf: No such file or directory

### ISSUE THREAD

**Issue by xiaozhazi** [xemul/criu#249]
I want to use criu to implement docker live migration. 
The source node has the same environment with destination node.
Docker version:
![image](https://cloud.githubusercontent.com/assets/4612045/20662132/0d6821e8-b58d-11e6-92a2-f70d57140c73.png)
criu version:
![image](https://cloud.githubusercontent.com/assets/4612045/20662145/20d05818-b58d-11e6-93ff-278001c93025.png)
sudo docker info:
![image](https://cloud.githubusercontent.com/assets/4612045/20662153/31ab296a-b58d-11e6-8570-6bb06b7cb4e6.png)
In the source node, run a docker container named lm_test1:
**sudo docker run -d --name lm_test1 busybox /bin/sh -c 'i=0; while true; do echo $i; i=$(expr $i + 1); sleep 1; done'**
I use the criu pre-dump and dump command to do the checkpoint. send the dump file to the destination.
And use criu restore the docker container:
**sudo criu restore --tree 51936 --images-dir /var/lib/docker/tmp/mobab1/dump --ext-mount-map auto**
it show the error message in restore.log as below:
![image](https://cloud.githubusercontent.com/assets/4612045/20662369/132463e8-b58e-11e6-8439-512c56a418f7.png)
Then I run the same command in the source node, it shows the error again.

![image](https://cloud.githubusercontent.com/assets/4612045/20662546/c6f40f22-b58e-11e6-8eac-8ed053cd1434.png)
![image](https://cloud.githubusercontent.com/assets/4612045/20662571/e8637cec-b58e-11e6-90b7-f0e394106bd9.png)

There has the two resolv.conf file. how can I solve the problem? 
and what does it means  Unable to statfs ./etc/resolv.conf: No such file or directory?

**Comment by xemul** [comment]
https://criu.org/External_bind_mounts

We have the live-migration engine sitting here https://github.com/xemul/p.haul
It's not very usable, but shows how it should work. And we have plans to move the p.haul's core into criu/ sources in the form of .go library.

So you're welcome to join :)

**Comment by xiaozhazi** [comment]
@xemul It's very excited for me to have opportunity to participate in this work. Thank you very much!
I know you have the p.haul to do the liv-migrate, so I wonder that may you have met this problem in restore the Docker container.  :)
I will see how p.haul works to find where is wrong in my live-migrate tools.

**Comment by xemul** [comment]
Actually, the dump and restore parts of live migration should be done using native docker API. In this case all the external mount points will be handled by Docker itself. So regarding your question about how many people hit this, the answer -- everybody that tries to use criu itself and nobody who uses docker API (that in turn calls criu with proper arguments) :)

**Comment by xiaozhazi** [comment]
I know the best way to recover a docker container is to use docker native C/R. If I use docker external C/R (use criu only), i can also finally restore the docker container, but docker daemon will not know the container is running again. so when use 'docker ps' may not get the information I want.

But I want to have a try, now question is I haven't restore the docker progress yet.

I found that is because I didn't use the proper command-opt, when I add '--manage-cgroups --evasive-devices --root XXXXX', this problem have been solved. 

But the restore function is not doing successfully yet. 
![image](https://cloud.githubusercontent.com/assets/4612045/20786307/1fc7c320-b7e0-11e6-8642-7e039572bb16.png)


But I think I can close this question now. Thank you for your advice about the proper arguments, It's really very helpful!
